sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -31,9 +31,9 @@ import org.apache.spark.sql.types.{AbstractDataType, DataType}
  *                  null. Use boxed type or [[Option]] if you wanna do the null-handling yourself.
  * @param dataType  Return type of function.
  * @param children  The input expressions of this UDF.
- * @param inputsNullSafe Whether the inputs are of non-primitive types or not nullable. Null values
- *                       of Scala primitive types will be converted to the type's default value and
- *                       lead to wrong results, thus need special handling before calling the UDF.
+ * @param inputPrimitives Whether the inputs are of primitive types. Null values of Java primitive
```

Comment:
=> The analyzer should be aware of Scala primitive types so as to make the UDF return null if there is any null input value of these types. On the other hand, Java UDFs can only have boxed types, thus this parameter will always be all false.

Reply:
+1